src/PekatVisionSDK/instance.py

```diff
@@ -0,0 +1,427 @@
+            pekat_install_base_name = "pekat-vision-"
+        else:
+            raise DistNotFoundError
+
+        installed_pekats = sorted(
+            pekat_install_parent_dir.glob(f"{pekat_install_base_name}*"),
+            key=lambda x: version.parse(x.name.split(pekat_install_base_name)[1]),
+            reverse=True,
+        )
+
+        if installed_pekats:
+            return installed_pekats[0]
+
+        raise DistNotFoundError
+
+    @staticmethod
+    def _find_free_ports() -> int:
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.bind(("localhost", 0))
+        _, port = s.getsockname()
+        s.close()
+        return port
+
+    @staticmethod
+    def _random_string(string_length: int) -> str:
+        letters = string.ascii_lowercase
+        return "".join(secrets.choice(letters) for _ in range(string_length))
+
+    def _check_project(self) -> bool:
+        if self.project_path is None:
+            return False
+        return (self.project_path / "pekat_package.json").exists()
+
+    def _start_instance(self) -> None:
+        if self.project_path is None:
+            raise ValueError(self.project_path)
+        if not self._check_project():
+            raise ProjectNotFoundError(self.project_path)
+
+        dist_path = self._get_dist_path()
+        server_path = dist_path / "pekat_vision" / "pekat_vision"
+        self.stop_key = self._random_string(10)
+
+        params = [
+            server_path,
+            "-data",
+            self.project_path,
+            "-port",
+            str(self.port),
+            "-host",
+            self.host,
+            "-stop_key",
+            self.stop_key,
+        ]
+
+        if self.gpu:
+            params += ["-gpu", str(self.gpu)]
+
+        # add other arguments
+        if self.disable_code:
+            params += ["-disable_code", self.disable_code]
+        if self.tutorial_only:
+            params += ["-tutorial_only", self.tutorial_only]
+
+        self.process = subprocess.Popen(
+            params,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+
+        stop_init_model = not self.wait_for_init_model
+        server_running = False
+
+        if self.process.stdout is None:
+            msg = "Process stdout is None"
+            raise RuntimeError(msg)
+
+        # wait for start
+        while True:
+            next_line = self.process.stdout.readline().decode()  # type: ignore[union-attr]
+            if next_line == "" and self.process.poll() is not None:
+                break
+            sys.stdout.flush()
+
+            if "__SERVER_RUNNING__" in next_line:
+                server_running = True
+            if "STOP_INIT_MODEL" in next_line:
+                stop_init_model = True
+            if server_running and stop_init_model:
+                return None
+
+            if next_line.find("OSError: [Errno 48] Address already in use") != -1:
```

Comment:
the `in` operator could be used here as well, like in lines 221 & 223

Reply:
Good point, fixed.